wallet/src/de/schildbach/wallet/ui/dashpay/EditProfileViewModel.kt

```diff
@@ -322,7 +322,6 @@ class EditProfileViewModel @Inject constructor(
                         config.imgurDeleteHash = imgurUploadResponse.data.deletehash
                         val avatarUrl = imgurUploadResponse.data.link
                         Log.d("AvatarUrl", avatarUrl)
-                        dashPayProfile?.avatarUrl = avatarUrl
```

Comment:
This doesn't seem to be needed (it's not used in the similar `uploadToGoogleDrive` method and doesn't affect the logic), but it's messing up later checks on whether the new avatar URL is different from the original. Let me know if I'm missing something.

Reply:
Good question.  Let me see.

Reply:
I believe that you are correct.  `dashPayProfile` contains the current profile data from the Room database and we shouldn't be changing any of its member fields.  It should only updated when the room database is updated.

If we change the image, then that image url is stored here `profilePictureUploadLiveData.value!!.data` after uploading.  When "Save" is pressed, then the updated profile with the new images or text is broadcast to the Platform Network and it is also saved to the Room database.